tests/UnitTests/CoreGraphics.drawing/CGContextDrawingTests.cpp

```diff
@@ -589,3 +589,181 @@ DRAW_TEST(CGContext, PremultipliedAlphaImage) {
     CGContextSetRGBFillColor(context, 1.0, 0.0, 0.0, 0.5);
     CGContextFillRect(context, { 0, 0, 100, 100 });
 }
+
+#pragma region LinearGradient
+DRAW_TEST_F(CGContext, LinearGradient, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+
+    CGFloat locations[2] = { 0, 1 };
+    CGFloat components[8] = { 0.0, 0.0, 1, 1.0, 1.0, 0, 0, 1.0 };
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(colorspace, components, locations, 2);
+
+    CGPoint startPoint = CGPointMake(0, 0);
+    CGPoint endPoint = CGPointMake(512, 1024);
+
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, kCGGradientDrawsBeforeStartLocation);
+    CFRelease(colorspace);
+    CFRelease(gradient);
+}
+
+static void _drawShortLinearGradientWithOptions(CGContextRef context, CGRect bounds, CGGradientDrawingOptions option) {
+    CGFloat locations[2] = { 0, 1 };
+    CGFloat components[8] = { 0.0, 1, 0.0, 1.0, 1.0, 0, 0, 1.0 };
+
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(colorspace, components, locations, 2);
+
+    CGRect borderRect = CGRectInset(bounds, 30, 50);
+
+    CGContextDrawLinearGradient(context, gradient, borderRect.origin, CGPointMake(borderRect.size.width, borderRect.size.height), option);
+
+    CFRelease(colorspace);
+    CFRelease(gradient);
+}
+
+DRAW_TEST_F(CGContext, LinearGradientShortBothSides_Options_0, UIKitMimicTest) {
+    _drawShortLinearGradientWithOptions(GetDrawingContext(), GetDrawingBounds(), 0);
+}
+
+DRAW_TEST_F(CGContext, LinearGradientShortBothSides_Options_kCGGradientDrawsBeforeStartLocation, UIKitMimicTest) {
```

Comment:
Each of these tests that resolves to a single function is better modelled as a parameterized test.